packages/@react-aria/numberfield/docs/useNumberField.mdx

```diff
@@ -0,0 +1,293 @@
+  ]} />
+
+## API
+
+<FunctionAPI function={docs.exports.useNumberField} links={docs.links} />
+
+## Features
+
+Number fields can be built with `<input type="number">`, but the behavior is very inconsistent across
+browsers and platforms, it supports limited localized formatting options, and it is challenging to style
+the stepper buttons. `useNumberField` helps achieve accessible number fields that support internationalized
+formatting options and can be styled as needed.
+
+* Support for internationalized number formatting and parsing including decimals, percentages, currency values, and units
+* Support for the Latin, Arabic, and Han decimal numbering systems in [over 30 locales](internationalization.html#supported-locales)
+* Automatically detects the numbering system used and supports parsing numbers not in the default numbering system for the locale
+* Support for multiple currency formats including symbol, code, and name in standard or accounting notation
+* Validates keyboard entry as the user types so that only valid numeric input according to the locale and numbering system is accepted
+* Handles composed input from input method editors, e.g. Pinyin
+* Automatically selects an appropriate software keyboard for mobile according to the current platform and allowed values
+* Supports rounding to a configurable number of fraction digits
+* Support for clamping the value between a configurable minimum and maximum, and snapping to a step value
+* Support for stepper buttons and arrow keys to increment and decrement the value according to the step value
+* Supports pressing and holding the stepper buttons to continuously increment or decrement
+* Handles floating point rounding errors when incrementing, decrementing, and snapping to step
+* Supports using the scroll wheel to increment and decrement the value
+* Exposed to assistive technology as a text field with a custom localized role description using ARIA
+* Follows the [spinbutton](https://www.w3.org/TR/wai-aria-practices-1.2/#spinbutton) ARIA pattern
+* Works around bugs in VoiceOver with the spinbutton role
+* Uses an ARIA live region to ensure that value changes are announced
+
+## Anatomy
+
+<Anatomy />
+
+Number fields consist of an input element that shows the current value and allows the user to type a new value,
+optional stepper buttons to increment and decrement the value, a group containing the input and stepper buttons,
+and a label.
+
+`useNumberField` returns props for each of these, which you should spread onto the appropriate elements:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.useNumberField.return.id].properties} />
+</TypeContext.Provider>
+
+If there is no visual label, an `aria-label` or `aria-labelledby` prop must be passed instead
+to identify the element to screen readers.
+
+## Example
+
+```tsx example export=true
+import {useNumberFieldState} from '@react-stately/numberfield';
```

Comment:
Something is broken with the examples:
![image](https://user-images.githubusercontent.com/9661127/106830786-985ce980-6643-11eb-9f5d-27c7b403e462.png)
![image](https://user-images.githubusercontent.com/9661127/106830848-ba566c00-6643-11eb-928a-5850d3ffa3ed.png)